task/buildah-oci-ta/0.2/buildah-oci-ta.yaml

```diff
@@ -429,16 +429,33 @@ spec:
         ACTIVATION_KEY_PATH="/activation-key"
         ENTITLEMENT_PATH="/entitlement"
 
-        # do not enable activation key and entitlement at same time. If both vars are provided, prefer activation key.
-        # when activation keys are used an empty directory on shared emptydir volume to "/etc/pki/entitlement" to prevent certificates from being included in the produced container
-        # To use activation key file 'org' must exist, which means the key 'org' must exist in the key/value secret
-
+        # 1. do not enable activation key and entitlement at same time. If both vars are provided, prefer activation key.
+        # 2. Activation-keys will be used when the key 'org' exists in the activation key secret.
+        # 3. try to pre-register and mount files to the correct location so that users do no need to modify Dockerfiles.
+        # 3. If the Dockerfile contains the string "subcription-manager register", add the activation-keys volume
+        #    to buildah but don't pre-register for backwards compatibility. In this case mount an empty directory on
+        #    shared emptydir volume to "/etc/pki/entitlement" to prevent certificates from being included in the produced
+        #    container.
+
+        REGISTERED="false"
         if [ -e /activation-key/org ]; then
           cp -r --preserve=mode "$ACTIVATION_KEY_PATH" /tmp/activation-key
           mkdir /shared/rhsm-tmp
           VOLUME_MOUNTS+=(--volume /tmp/activation-key:/activation-key -v /shared/rhsm-tmp:/etc/pki/entitlement:Z)
           echo "Adding activation key to the build"
 
+          if ! grep subscription-manager register "$dockerfile_path"; then
+            # user is not running registration in the Containerfile: pre-register.
+            echo "Pre-registering with subscription manager."
+            subscription-manager register --org "$(cat /tmp/activation-key/org)" --activationkey "$(cat /tmp/activation-key/activationkey)"
+            REGISTERED=$?
+            # copy generated certificates to /shared/rhsm-tmp
+            cp /etc/pki/entitlement/*.pem /shared/rhsm-tmp
+
+            # and then mount get /etc/rhsm/ca/redhat-uep.pem into /run/secrets/rhsm/ca
+            VOLUME_MOUNTS+=(--volume /etc/rhsm/ca/redhat-uep.pem:/run/secrets/rhsm/ca/redhat-uep.pem)
+          fi
```

Comment:
So if I understand correctly:
* we are generating entitlement certificates in a Fedora container image (the one that runs this script)
* the Fedora image installs `subscription-manager`, which comes with the `redhat-uep.pem` file
* we're mounting those into the build, which could be based on ubi8, ubi9, ubiX-minimal, ...

@ptoscano could you check if we're doing something sensible here?

Reply:
I _think_ that the redhat-uep.pem is actually generated during the registration, but the rest of what you say is accurate

Reply:
> I _think_ that the redhat-uep.pem is actually generated during the registration,

```bash
$ rpm -ql subscription-manager-rhsm-certificates 
/etc/rhsm
/etc/rhsm/ca
/etc/rhsm/ca/redhat-entitlement-authority.pem
/etc/rhsm/ca/redhat-uep.pem
/usr/share/licenses/subscription-manager-rhsm-certificates
/usr/share/licenses/subscription-manager-rhsm-certificates/COPYING
```

Reply:
@chmeliik other than what I wrote above about "where redhat-uep.pem comes from", what you said seems correct

Reply:
And can it cause some problems? Mounting `redhat-uep.pem` from the Fedora version of the package? Or does it generally not matter?

Reply:
It should not be an issue, since `redhat-uep.pem` is shipped by `subscription-manager-rhsm-certificates`, which would need to be upgraded in case any of the certificates it ships needs to be changed. In that case, the update of the package ought to be done enough time in advance to ensure all the users (RHEL and not only, Fedora included) get the new package installed.